### PR TITLE
Adding three tuning techniques using GA, EDA and PSO

### PR DESCRIPTION
**Pull request by rgmantovani**
Three optimization methods for hyper-parameter tuning. All of them just handle numeric and integer hyper-parameters:
- **tuneGA** (using [GA](https://cran.r-project.org/web/packages/GA/index.html) R package): a real-valued Genetic Algorithm. User can set some of the GA's parameters, such as: mutation probability, crossover probability, population size and the number of generations.
- **tunePSO** (using [pso](https://cran.r-project.org/web/packages/pso/index.html) R package ): a standard real-valued Particle Swarm Optimization. User can set parameters, such as: the PSO type (2007 or 2011 standard), the number of swarm particles and the number of generations.
- **tuneEDA** (using [copulaedas](https://cran.r-project.org/web/packages/copulaedas/index.html) R package ): some Estimation of Distribution Algorithms that use copula and vine functions. User can specify which EDA he would like to use: UMDA, GCEDA, CVEDA or DVEDA. It is also possible to define the number of iterations and population size.

### CODE REVIEW

**Comment by larskotthoff** [comment]
Thanks, looks good (apart from the documentation issues). Will merge for 2.6.

**Comment by berndbischl** [comment]
there seem to be some irrelvant (?) warnings?
can those be treated?

I would also like to review this a final time before we merge in 2.6

**Comment by larskotthoff** [comment]
The warnings are because of the documentation, as discussed yesterday (I'm having a look at this now).

I'll only merge this after 2.5 has been released. So you'll have plenty of time to review...

**Comment by berndbischl** [comment]
ok thx.

that with the review was mainly a not for later so we dont forget

**Comment by larskotthoff** [comment]
Ok, I've rebased on master after fixing the documentation. Build succeeds now.

**Comment by rgmantovani** [comment]
Thank you.

**Comment by larskotthoff** [comment]
@berndbischl Do you still want to include that in 2.9?

**Comment by larskotthoff** [comment]
@berndbischl Should we merge this?

**Comment by berndbischl** [comment]
> @berndbischl Should we merge this?

well have you properly reviewed this? whats your opinion here?

we also would have to rebase this on master, seems to have conflicts

**Comment by larskotthoff** [comment]
Looks good to me. Rebased now.

**Comment by jakob-r** [comment]
The code could benefit from `BBmisc::coalesce()`

**Comment by mllg** [comment]
Wow, this needs a rebase again. But looks good to me too.

**Comment by mllg** [comment]
I've cleaned up the code a bit and also found 1-2 errors (maybe because mlr changed something internally). Everything should be fine now, hopefully Travis and Jenkins agree...

**Comment by berndbischl** [comment]
@mllg @rgmantovani 
Michel, thanks a lot for taking this over, this was really needed!

Rafael, can you proof read this at some point pls?

**Comment by mllg** [comment]
Note that the modifications on NEWS are not in this PR anymore but can be retrieved from Rafael's initial commit after this one is merged.

File: R/tuneEDA.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  if (is.null(start))
+    start = sampleValue(par.set, start, trafo = FALSE)
+  start = convertStartToNumeric(start, par.set)
+
+  ctrl.eda = list(popSize = 100L, maxit = 50L, type = "UMDA")
+  ctrl.eda = insert(ctrl.eda, control$extra.args)
+  cx = function(x, par.set) convertXNumeric(x, par.set)
+
+  if (is.null(ctrl.eda$type) || ctrl.eda$type %nin% c("UMDA", "GCEDA", "CVEDA", "DVEDA"))
+    ctrl.eda$type = "UMDA"
```
Comment (ja-thomas) [inline R/tuneEDA.R]: This seems a bit strange, we fallback to the default if the user specifies nothing or a "wrong" type. why not just setting a default and doing a check if the user sets something stupid and throw an error

**Comment by rgmantovani** [comment]
@berndbischl Sorry for the delay. What is still missing from my side ?

File: R/tuneEDA.R (hunk 1 above)
Comment (rgmantovani) [inline R/tuneEDA.R]: @ja-thomas which one should be the default? the simplest one?

**Comment by berndbischl** [comment]
@rgmantovani 

please do the following:

1) address what @ja-thomas has posted. if the underlying package does define a default, pick that one.

2) please carefully proofread yoir code again. and read check it against the mlr devel guidelines. 
after that i will check again and then merge

File: R/tuneEDA.R (hunk 1 above)
Comment (rgmantovani) [inline R/tuneEDA.R]: Well, done! Methods have defaults and throw error.

**Comment by rgmantovani** [comment]
I did the changes but didn't commit yet. 
I have one *problem* here running the unit tests for EDAs:
* just once: OK
* twice: crash

If I close the R session and run it again works. The message I see is this one:
```
Error in .Call("do_ptruncnorm", q, a, b, mean, sd) : 
  "do_ptruncnorm" not resolved from current namespace (truncnorm)
```

The call is done by ```truncnorm``` package (used by ```copulaedas```):
https://github.com/cran/truncnorm/blob/master/R/truncnorm.R

A similar error was described here:
https://github.com/hadley/plyr/issues/258
http://stackoverflow.com/questions/31352789/keep-hitting-the-error-loop-apply-not-resolved-from-current-namespace-plyr

Does anyone have any ideia or tip to give about it?

screen shot 2016-09-22 at 9 34 37 pm

**Comment by rgmantovani** [comment]
@berndbischl I've forked the dir again so I can't update here. Can I close this PR and open a new one?

**Comment by rgmantovani** [comment]
@berndbischl, I added the defaults and check the files.
Please, let me know if I need to change something else.

**Comment by larskotthoff** [comment]
Looks like this isn't going to happen, and we have MBO tuning now. I'm closing this, but please reopen if you feel that this is wrong.

**Pull request closed without merging**